### PR TITLE
Add `invalidReasons` to ImageResponse

### PR DESCRIPTION
**Pull request by kenoir**
In order that the UI can provide a more granular indicator of validity.

screenshot from 2016-04-19 15 41 50

Upload screen changes:
screenshot from 2016-04-19 16 27 04

### CODE REVIEW

**Comment by kenoir** [comment]
@Reettaphant ping!

**Comment by ShaunYearStrong** [comment]
👍

File: media-api/app/lib/ImageExtras.scala [hunk 1]
```diff
@@ -7,6 +7,34 @@ import com.gu.mediaservice.model._
...
+    o.map(_ => true).getOrElse(false)
+
+  def hasCredit(meta: ImageMetadata) = optToBool(meta.credit)
+  def hasDescription(meta: ImageMetadata) = optToBool(meta.description)
+  def isPng(asset: Asset) = asset.mimeType == Some("image/png")
+
+  def validityMap(image: Image): Map[String, Boolean] = Map(
+    "has_credit" -> hasCredit(image.metadata),
+    "has_description" -> hasDescription(image.metadata),
+    "is_png" -> !isPng(image.source)
```
Comment (ShaunYearStrong) [inline media-api/app/lib/ImageExtras.scala]: Bit confusing here, `is_png` is not a png?

File: media-api/app/lib/elasticsearch/SearchFilters.scala [hunk 2]
```diff
@@ -13,12 +13,11 @@ import lib.Config
 
 trait SearchFilters extends ImageFields {
 
+  import UsageRightsConfig.{ suppliersCollectionExcl, freeSuppliers }
+
   val validFilter   = Config.requiredMetadata.map(metadataField).toNel.map(filters.exists)
```
Comment (ShaunYearStrong) [inline media-api/app/lib/elasticsearch/SearchFilters.scala]: This is now decoupled from imageextras def of validity.

**Pull request merged**